secant/Screens/Onboarding/OnboardingScreenViewModel.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  OnboardingScreenViewModel.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 9/17/21.
+//
+
+import Foundation
+import Combine
+
+protocol OnboardingStepProviding {
```

Comment:
Is this protocol providing an abstraction that is used anywhere?  I don't see this being used more than once and I also don't see this being leveraged in tests, I think we might be able to remove it and just work with value types.

Reply:
This is not definitive though. We still have to define how to approach the real onboarding resources. The mock started to get real and I moved it over to this target. I can delete this protocol if it's strictly necessary. I assume that we will be adding it sooner than later.

Reply:
What I don't see here though is a reason for this to be a protocol.  I guess maybe this is the hard part about reviewing incomplete code?  I'm not seeing how this is something that needs to be generic vs just using a value.  If you think this is going to be needed later then it is what it is but I just don't have any reason to currently see the value of this protocol for it's usage or for testing.

Reply:
I think you are right, but I also think that we are in the initial phases of development where some parts of the code will leave room for near-future enhancements.

For example, protocols, are very good at capturing and code-documenting requirements. When you approach an unknown codebase, protocols immediately tell you what the important things are. They also help you not to be second guessing abstractions when you want to re-implement or refactor things. Coding to interfaces adds value even though there's one implementation for it. 

On TDD, if you start with a protocol, you've kill two birds with one stone since LoC number 1. You program your tests to that protocol, your mock it and go down the red-green-refactor cycle, then you code the SUT and implement the real protocol, on the other side. Sometimes as it is this case, things are simple and due to the poor mocking tools in place, mocks are really similar to the real implementations. That doesn't mean that the protocol did not add value. Probably the opposite, it's the reason you got there.

Reply:
I can see the appeal of wanting to work with protocols if the company was practicing TDD.  It is common though to optimize for for theoretical concerns.  I do see the value in adding protocols, and working with abstractions, there are many benefits as you've pointed out.  I'm not as in tune with the future of this app yet and I'm not as involved in the early architecture planning as your are but my advice would be to not push the code into an abstraction for the sake of doing so, but rather implement the code as needed and discover the abstractions as they are needed.  This is somewhat similar to gathering requirements before starting to code.  Adding protocols that don't actually provide any type of abstraction and that aren't used to help testing in any way, could make the code harder to work with in the future if or when you do discover a valuable abstraction to utilize.  I'll defer to your judgment though since I'm less apprised to what's coming next.